### PR TITLE
LibWeb: Stop radio buttons firing change events when losing focus

### PR DESCRIPTION
**Pull request by Psychpsyo**
I have checked since #2379 and radio buttons should indeed work this way also.

### CODE REVIEW

File: Tests/LibWeb/Text/input/radio-button-focus-lost-no-change-event.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE html>
+<input type="radio" id="radioButton">
+<script src="include.js"></script>
+<script>
+    radioButton.addEventListener("change", () => {
+      println("Change event was fired when it shouldn't have been.");
```
Comment (tcl3) [inline Tests/LibWeb/Text/input/radio-button-focus-lost-no-change-event.html:6]: Can we change the test so that it prints something if the test passes?

Something like:
```diff
diff --git a/Tests/LibWeb/Text/input/radio-button-focus-lost-no-change-event.html b/Tests/LibWeb/Text/input/radio-button-focus-lost-no-change-event.html
index 1195b4d0672..f595c7af56f 100644
--- a/Tests/LibWeb/Text/input/radio-button-focus-lost-no-change-event.html
+++ b/Tests/LibWeb/Text/input/radio-button-focus-lost-no-change-event.html
@@ -2,15 +2,20 @@
 <input type="radio" id="radioButton">
 <script src="include.js"></script>
 <script>
+    let changeEventFired = false;
     radioButton.addEventListener("change", () => {
-      println("Change event was fired when it shouldn't have been.");
+        changeEventFired = true;
     });
     asyncTest(async done => {
         radioButton.focus();
         await new Promise(resolve => setTimeout(resolve, 0));
         radioButton.blur();
         await new Promise(resolve => setTimeout(resolve, 0));
-
+        if (changeEventFired) {
+            println("FAIL: Change event was fired when it shouldn't have been.");
+        } else {
+            println("PASS: Change event was not fired");
+        }
         done();
     });
-</script>
\ No newline at end of file
+</script>
```

File: Tests/LibWeb/Text/input/radio-button-focus-lost-no-change-event.html [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    });
+    asyncTest(async done => {
+        radioButton.focus();
+        await new Promise(resolve => setTimeout(resolve, 0));
+        radioButton.blur();
+        await new Promise(resolve => setTimeout(resolve, 0));
+
+        done();
+    });
+</script>
```
Comment (tcl3) [inline Tests/LibWeb/Text/input/radio-button-focus-lost-no-change-event.html:16]: Newline at end of file, please.

File: Tests/LibWeb/Text/input/radio-button-focus-lost-no-change-event.html (hunk 1 above)
Comment (Psychpsyo) [reply Tests/LibWeb/Text/input/radio-button-focus-lost-no-change-event.html:6]: Yes, should I change the equivalent `checkbox-focus-lost-no-change-event.html` in this same PR?
(the two tests look almost identical, I just copied and modified the checkbox one to make this one)

Comment (tcl3) [reply Tests/LibWeb/Text/input/radio-button-focus-lost-no-change-event.html:6]: > Yes, should I change the equivalent `checkbox-focus-lost-no-change-event.html` in this same PR?

Yes please, that should be fine as a separate commit in this PR.

Comment (Psychpsyo) [reply Tests/LibWeb/Text/input/radio-button-focus-lost-no-change-event.html]: Done.

File: Tests/LibWeb/Text/input/radio-button-focus-lost-no-change-event.html (hunk 2 above)
Comment (Psychpsyo) [reply Tests/LibWeb/Text/input/radio-button-focus-lost-no-change-event.html]: Added.

File: Tests/LibWeb/Text/input/checkbox-focus-lost-no-change-event.html [hunk 3]
```diff
@@ -2,15 +2,22 @@
...
+            println("FAIL: Change event was fired when it shouldn't have been.");
+        } else {
+            println("PASS: Change event was not fired");
+        }
+
         done();
     });
-</script>
\ No newline at end of file
+</script>
```
Comment (tcl3) [inline Tests/LibWeb/Text/input/checkbox-focus-lost-no-change-event.html:23]: Can you put these changes in a separate commit, please.

Comment (Psychpsyo) [reply Tests/LibWeb/Text/input/checkbox-focus-lost-no-change-event.html]: Yes, done, sorry.

**Review by tcl3 — APPROVED** [review]
